python/pyspark/sql/connect/group.py

```diff
@@ -262,8 +264,44 @@ def applyInPandas(
 
     applyInPandas.__doc__ = PySparkGroupedData.applyInPandas.__doc__
 
-    def applyInPandasWithState(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("applyInPandasWithState() is not implemented.")
+    def applyInPandasWithState(
+            self,
+            func: "PandasGroupedMapFunctionWithState",
+            outputStructType: Union[StructType, str],
```

Comment:
I see that other part of the code use quoted types, i.e. `Union["StructType", str]`. Maybe we should also do that for code consistency? 
Also doing that seems to [help with forward consistency](https://stackoverflow.com/questions/46458470/should-you-put-quotes-around-type-annotations-in-python)

Reply:
The [current function](https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/group_ops.py#L233) uses `Union[StructType, str]`. Also, majority of places use this in other files.

Reply:
Using quotes is only necessary when the type cannot be imported property because of cyclic references or if the type is not defined yet.